### ISSUE TITLE
Final brain template image cut off using Buildtemplateparallel.sh

### ISSUE THREAD

**Issue by mahabbat410** [ANTsX/ANTs#1325]
Hi,

I am trying to make T1 Brain template using buildtemplateparallel.sh command.
I have successfully managed to run the code with no errors, but my final brain template seems to be cut off.
![Final Brain template](https://user-images.githubusercontent.com/92302079/159110564-b0ce2a54-aa3f-43bf-9f08-fce0aee2d85e.png)
I’ve checked all images produced from running this code (Warp, InverseWarp, repaired and Affine.txt), and the warp image and inverse warp images seems to be cut off in the same way as the final brain template.
![Original Warp Image](https://user-images.githubusercontent.com/92302079/159110741-0d713fff-48fb-41fd-ae10-b5ab83f16e95.png)
I think it is related to editing the field of view but can’t seem to find any options to edit FOV.

I’ve also tried to run the examples provided in https://github.com/ntustison/TemplateBuildingExample, which works absolutely fine.

Do you have any idea what can I do to fix this?

The command line I’ve used is:
buildtemplateparallel.sh -d 3 -o NonlinearCC -c 2 -j 30 -n 0 -t RA -m 10000x10000x10000x10000 -s CC -t GR -m 100x100x100 -s CC -z Affine_BTP.nii.gz input images.nii.gz

[my reference brain template is made from previous BTP command using Affine registration of the same subject, and I’ve also performed N4Bias correction in preprocessing, hence skipped this process here]

Many thanks,

**Comment by ntustison** [comment]
``buildtemplateparallel.sh`` was supplanted by ``antsMultivariateTemplateConstruction.sh`` and ``antsMultivariateTemplateConstruction2.sh``.  In these latter two scripts, you can set ``-y 0`` which has a good chance of solving this issue.

**Comment by gdevenyi** [comment]
@ntustison maybe time to add a stop and print a warning chunk of code with an override?

**Comment by ntustison** [comment]
Yeah, you're probably right.  What specifically did you have in mind with the override?

**Comment by mahabbat410** [comment]
Thank you so much for your advice,
I've now changed to antsMultivariateTemplateConstruction.sh to make the template with the option -y 0, but still have the same problem.
The final template warp image seems to cover the whole area
![image](https://user-images.githubusercontent.com/92302079/159165159-347f38b1-7462-41a8-aa39-8cc13de8dcf0.png)

but the individual warp images show this
![image](https://user-images.githubusercontent.com/92302079/159165175-0b96a318-88ec-4d04-b61e-9e2469329378.png)

the changed code is:
antsMultivariateTemplateConstruction.sh -d 3 -o NonlinearCC -c 2 -j 30 -n 0 -t RA -m 10000x10000x10000x10000 -s CC -t GR -m 100x100x100 -z Affine_BTP.nii.gz -y 0  <input images>

I've checked all the images (input images, the reference image, and the final template), and they were in the same orientation except the individual warp images (as the image above). Are there any options to manually assign warp images to certain coordinates?

Many thanks

**Comment by gdevenyi** [comment]
What does Affine_BTP.nii.gz look like?

**Comment by mahabbat410** [comment]
the final template for Affine looks like this
![image](https://user-images.githubusercontent.com/92302079/159169378-494bedb5-111e-4f71-9fb1-591f60ab1f54.png)

And the warp images are similar to the images above. (individual warp images are displayed in different orientations, but the final warp image covers the whole area - below is an individual warp image with Affine_BTP.nii.gz)
![image](https://user-images.githubusercontent.com/92302079/159169599-f5536b95-fd93-401b-92dc-b44c55dfbaa6.png)

I've also changed the affine template to antsMultivariateTemplateConstruction.sh (with -y 0 options) after the nonlinear registration still showing problems and the results are the same as the second image.

**Comment by cookpa** [comment]
I suspect something in the image headers. We need to know your ANTs version because the interpretation of NIFTI headers in ITK has changed over time.

It would be helpful if you could share a reproducible exampling including Affine_BTP.nii.gz, a subset of your input images (say 3) and a command that produces the above problem.

**Comment by mahabbat410** [comment]
I am currently using ANTs Version: 2.3.4.

And here are the template and the subset of the data
[test_subj1.nii.gz](https://github.com/ANTsX/ANTs/files/8313668/test_subj1.nii.gz)
[test_subj2.nii.gz](https://github.com/ANTsX/ANTs/files/8313669/test_subj2.nii.gz)
[test_subj3.nii.gz](https://github.com/ANTsX/ANTs/files/8313670/test_subj3.nii.gz)
[Affine_BTP.nii.gz](https://github.com/ANTsX/ANTs/files/8313772/Affine_BTP.nii.gz)

and the command line used:
antsMultivariateTemplateConstruction.sh -d 3 -o NonlinearCC/NonlinearCC -c 2 -j 30 -n 0 -t RA -m 10000x10000x10000x10000 -s CC -t GR -m 100x100x100 -z Affine_BTP.nii.gz -y 0  test_subj*.nii.gz

I've realized that as I was switching from buildtemplateparallel.sh to antsMultivariateTemplateConstruction.sh I didn't change the transformation modalities (RA). Would this affect the result? If so, how can I perform affine registration using antsMultivariateTemplateConstruction.sh?

**Comment by cookpa** [comment]
Thanks. The headers all look reasonable.

Perhaps a manifestation of #1076? You can see if the latest ANTs fixes it. 

There's not an option for Affine only in antsMultivariateTemplateConstruction.sh but you can do `-t GR -m 1x0x0` to get a close to affine solution. antsMultivariateTemplateConstruction2.sh, which uses antsRegistration, lets you chose an affine-only or rigid-only transform.

**Comment by cookpa** [comment]
Quick test:

```
antsMultivariateTemplateConstruction.sh -d 3 -o NonlinearMI/NonlinearMI -c 2 -j 3 -n 0 -t GR -m 100x20x0 -s MI -z Affine_BTP.nii.gz -y 0 test_subj*.nii.gz
```

<img width="1061" alt="image" src="https://user-images.githubusercontent.com/611507/159270450-464dd191-72a1-4537-93e5-aaafe64250a8.png">

Results are obviously suboptimal but I don't see the problem. I think you have a bug in your version of ANTS that is fixed in v2.3.5

**Comment by mahabbat410** [comment]
Thank you so much for your help! I've upgraded to version 2.3.5 and it worked!
